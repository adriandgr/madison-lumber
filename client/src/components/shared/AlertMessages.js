import React from 'react';
import PropTypes from 'prop-types';
import scroll from '../../utils/scrollTo';

const AlertMessages = props => {
  let msgCount = 0;
  let infoCount = 0;
  let errCount = 0;

  if((props.success.length > 0 || props.errors.length > 0 || (props.info && props.info.length > 0))  && props.scroll) {
    scroll.to('.alert-messages');
  }
  return (
  <div className="alert-messages">
    { props.success.length > 0 &&
      <div className="alert alert-success">
        {props.success.map((msg, i) => {
          msgCount++;
          return (
          <span key={`msg${msgCount}`}>
            { i === 0
              ? <strong>{msg} <br/></strong>
              : <span>{msg} <br/></span> }
          </span>
          )
        })}
      </div>
    }

    { props.info && props.info.length > 0 &&
      <div className="alert alert-info">
        {props.info.map((info, i) => {
          infoCount++;
          return (
          <span key={`info${infoCount}`}>
            { i === 0
              ? <strong><i className="fa fa-info-circle" aria-hidden="true"></i> {info} <br/></strong>
              : <span>{info} <br/></span> }
          </span>
          )
        })}
      </div>
    }

    { props.errors.length > 0 &&
      <div className="alert alert-danger">
        {props.errors.map((err, i) => {
          errCount++;
          return (
          <span key={`err${errCount}`}>
            { i === 0
              ? <strong>{err} <br/></strong>
              : <span>{err} <br/></span> }
          </span>
          )
        })}
      </div>
    }
  </div>
)};

AlertMessages.propTypes = {
  success: PropTypes.array.isRequired,
  errors: PropTypes.array.isRequired
};

export default AlertMessages;