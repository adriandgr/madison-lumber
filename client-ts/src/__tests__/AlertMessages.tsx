import React from 'react';
import { shallow, mount } from 'enzyme';
import AlertMessages from '../components/shared/AlertMessages';
import renderer from 'react-test-renderer';


const successMsg = ['Success', 'This works...']
const errorMsg = ['Error!', 'This displays something bad...']
const infoMsg = ['Info', 'This is a true fact...']

it('renders without crashing', () => {
  shallow(<AlertMessages />);
});

it('returns null when no props are passed', () => {
  const component = renderer.create(<AlertMessages />);
  let tree = component.toJSON();
  expect(tree).toBeNull();
});

it('renders success message correctly', () => {
  const component = renderer.create(
    <AlertMessages success={successMsg} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders error message correctly', () => {
  const component = renderer.create(
    <AlertMessages error={errorMsg} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders info message correctly', () => {
  const component = renderer.create(
    <AlertMessages info={infoMsg} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders mixed type message correctly', () => {
  const component = renderer.create(
    <AlertMessages
      success={successMsg}
      error={errorMsg}
      info={infoMsg} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});