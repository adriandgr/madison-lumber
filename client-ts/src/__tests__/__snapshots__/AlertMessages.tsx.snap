// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error message correctly 1`] = `
<div
  className="alert-messages"
>
  <div
    className="alert alert-danger"
  >
    <span>
      <strong>
        Error!
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This displays something bad...
         
        <br />
      </span>
    </span>
  </div>
</div>
`;

exports[`renders info message correctly 1`] = `
<div
  className="alert-messages"
>
  <div
    className="alert alert-info"
  >
    <span>
      <strong>
        <i
          aria-hidden="true"
          className="fa fa-info-circle"
        />
         
        Info
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This is a true fact...
         
        <br />
      </span>
    </span>
  </div>
</div>
`;

exports[`renders mixed type message correctly 1`] = `
<div
  className="alert-messages"
>
  <div
    className="alert alert-success"
  >
    <span>
      <strong>
        Success
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This works...
         
        <br />
      </span>
    </span>
  </div>
  <div
    className="alert alert-info"
  >
    <span>
      <strong>
        <i
          aria-hidden="true"
          className="fa fa-info-circle"
        />
         
        Info
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This is a true fact...
         
        <br />
      </span>
    </span>
  </div>
  <div
    className="alert alert-danger"
  >
    <span>
      <strong>
        Error!
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This displays something bad...
         
        <br />
      </span>
    </span>
  </div>
</div>
`;

exports[`renders success message correctly 1`] = `
<div
  className="alert-messages"
>
  <div
    className="alert alert-success"
  >
    <span>
      <strong>
        Success
         
        <br />
      </strong>
    </span>
    <span>
      <span>
        This works...
         
        <br />
      </span>
    </span>
  </div>
</div>
`;
